src/Adapter/Order/CommandHandler/AddCartRuleToOrderHandler.php

```diff
@@ -65,12 +66,33 @@ public function handle(AddCartRuleToOrderCommand $command): void
         }
 
         $cartRules = [];
-        $discountValue = (float) str_replace(',', '.', $command->getDiscountValue());
+        $discountValue = (float) (string) $command->getDiscountValue();
```

Comment:
I'm not a big fan of double cast without comment.

Reply:
It's used in a lot of handlers that use `Decimal\Number` Sadly, if we add a comment each time this is used the codebase is gonna be crowded pretty quick
That's why I wanted to update the library https://github.com/PrestaShop/decimal/issues/6
But as @eternoendless mentioned, the core issue is that we mix float and Number for now